File: product-management-api/Models/Product.cs
namespace ProductManagementApi.Models;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Status { get; set; } = "active";
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: product-management-web/src/types/product.ts
import type { IPaginatedResponse } from "@/types/pagination"

export type ProductStatus = 'active' | 'inactive' | 'deleted'

export interface IProduct {
  id: number
  name: string
  status: ProductStatus
  created_at: string
  updated_at: string | null
}

export interface IProductsResponse extends IPaginatedResponse {
  products: IProduct[]
}
